python/tank/templatekey.py

```diff
@@ -13,6 +13,8 @@
 """
 
 import re
+import datetime as dt
```

Comment:
I always found it realy confusing when libraries are renamed like this. I suggest we do 

```
import datetime # for datetime.time, datetime.date
from datetime import datetime # to avoid datetime.datetime
```

Curious to hear what you think about that approach.